migrations/migrationUtils/helper.js

```diff
@@ -8,118 +8,186 @@ This module provides utility functions for interacting with Kafka, performing pr
+      },
+      body: JSON.stringify({
         identifier: IDENTIFIER,
         password: PASSWORD,
       }),
-      {
-        headers: {
-          Origin: ORIGIN,
-          'Content-Type': 'application/json',
-        },
+    };
+
+    request.post(options, (err, res, body) => {
+      if (err) {
+        console.error('[Auth Error] Failed to login and fetch token:', err.message);
+        return reject(err);
       }
-    );
-    console.log('[Auth Success] Token fetched successfully.');
-    return response.data.result;
-  } catch (err) {
-    console.error('[Auth Error] Failed to login and fetch token:', err.response?.data || err.message);
-    throw err;
-  }
+
+      try {
+        const data = typeof body === 'object' ? body : JSON.parse(body);
+
+        if (res.statusCode === 200 && data?.result) {
+          console.log('[Auth Success] Token fetched successfully.');
+          return resolve(data.result);
+        } else {
+          console.error('[Auth Error] Unexpected response:', data);
+          return reject(new Error('Invalid login response'));
+        }
+      } catch (parseErr) {
+        console.error('[Auth Error] Failed to parse response:', parseErr.message);
+        return reject(parseErr);
+      }
+    });
+  });
 }
 
 // Push completed survey submission
 async function pushCompletedSurveySubmissionToKafka(id, token, DOMAIN, completedSurveyPushEndpoint) {
-  try {
+  return new Promise((resolve, reject) => {
     console.log(id, 'id in pushCompletedSurveySubmissionToKafka');
-    const response = await axios.post(
-      `${DOMAIN}${completedSurveyPushEndpoint}/${id}`,
-      {},
-      {
-        headers: {
-          'Content-Type': 'application/json',
-          Accept: 'application/json, text/plain, */*',
-          'x-auth-token': token,
-        },
+
+    const url = `${DOMAIN}${completedSurveyPushEndpoint}/${id}`;
+    const options = {
+      url,
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json, text/plain, */*',
+        'x-auth-token': token,
+      },
+      json: {}, // empty body, same as axios.post({}, ...)
+    };
+
+    request.post(options, (err, res, body) => {
+      if (err) {
+        console.error(`[Failed] Survey submission ${id} failed to push:`, err.message);
+        return reject(err);
       }
-    );
-    console.log(response.data, 'response from pushCompletedSurveySubmissionToKafka');
-  } catch (err) {
-    console.error(`[Failed] Survey submission ${id} failed to push:`, err.response?.data || err.message);
-  }
-}
 
+      try {
+        const data = typeof body === 'object' ? body : JSON.parse(body);
+        console.log(data, 'response from pushCompletedSurveySubmissionToKafka');
+        resolve(data);
+      } catch (parseErr) {
+        console.error(`[Failed] Survey submission ${id} response parse error:`, parseErr.message);
+        reject(parseErr);
+      }
+    });
+  });
+}
 const projectServiceProgramUpdate = function ({
```

Comment:
@borkarsaish65 do we have to keep this here. if not used remove this

Reply:
done